subsys/bluetooth/audio/csip_set_member.c

```diff
@@ -81,8 +82,23 @@ struct bt_csip_set_member_svc_inst {
 };
 
 static struct bt_csip_set_member_svc_inst svc_insts[CONFIG_BT_CSIP_SET_MEMBER_MAX_INSTANCE_COUNT];
+static struct k_mutex svc_inst_lock;
 
 static void deferred_nfy_work_handler(struct k_work *work);
+static void add_bonded_addr_to_client_list(const struct bt_bond_info *info, void *data);
+
+#if defined(CONFIG_BT_SETTINGS)
+static int csip_settings_commit(void)
+{
+	bt_foreach_bond(BT_ID_DEFAULT, add_bonded_addr_to_client_list, NULL);
+
+	LOG_DBG("Restored CSIP client list from bonded devices");
+
+	return 0;
+}
+
+BT_SETTINGS_DEFINE(csip_set_member, "csip", NULL, csip_settings_commit);
```

Comment:
Have you tested this to work? 

Looking at the code in `keys.c`, it looks like the keys are read from settings with the same priority as here: possible race condition?

We may need to use `BT_SETTINGS_CPRIO_2` or above here